Add aliases for many GNU long options and add `SHOPT_ALL_LIBCMD`

JohnoKing opened the pull request:
Changes:
- The following long option aliases have been added to the libcmd builtins (to match the equivalent options in GNU coreutils):
`basename --multiple` == `basename -a`
`cat --show-tabs` == `cat -T`
`chgrp --no-dereference` == `chgrp -h`
`chown --no-dereference` == `chown -h`
`cp --no-dereference` == `cp -P`
`join --ignore-case` == `join -i`
~`ln --symbolic` == `ln -s` (because of shared code, this also adds `--symbolic` to `cp` and `mv`)~
`rm --dir` == `rm -d`

- Added `kill --signal` as an alias for `kill -s` (matches the equivalent long option in util-linux).

- A new `SHOPT_ALL_LIBCMD` option (which is off by default) has been added that enables all of the builtins in src/lib/libcmd (with the exception of `vmstate` if vmalloc isn't enabled). This is primarily to make testing the /opt/ast/bin builtins a bit easier.

- Fixed an error in the documentation of `cat -T`. This option does not display formfeeds as ^L. No change in behavior is needed as the current behavior matches that of the GNU cat command.

- Fixed various minor issues in the documentation for the `kill` builtin.

JohnoKing committed:
Add aliases for many GNU long options and add SHOPT_ALL_LIBCMD

Changes:
- The following long option aliases have been added to the libcmd
  builtins (to match the equivalent options in GNU coreutils):
  'basename --multiple' == 'basename -a'
  'cat --show-tabs' == 'cat -T'
  'chgrp --no-dereference' == 'chgrp -h'
  'chown --no-dereference' == 'chown -h'
  'cp --no-dereference' == 'cp -P'
  'join --ignore-case' == 'join -i'
  'ln --symbolic' == 'ln -s' (because of shared code, this also adds
                              --symbolic to cp and mv)
  'rm --dir' == 'rm -d'

- Added 'kill --signal' as an alias for 'kill -s' (matches the
  equivalent long option in util-linux).

- A new SHOPT_ALL_LIBCMD option (which is off by default) has been added
  that enables all of the builtins in src/lib/libcmd (with the exception
  of vmstate if vmalloc isn't enabled). This is primarily to make
  testing the /opt/ast/bin builtins a bit easier.

- Fixed an error in the documentation of 'cat -T'. This option does not
  display formfeeds as ^L. No change in behavior is needed as the
  current behavior matches that of the GNU cat command.

- Fixed various minor issues in the documentation for the kill builtin.

JohnoKing committed:
Backport ksh2020 regression tests for various builtins

This commit backports many regression tests from ksh2020 for the
following builtins:
  /opt/ast/bin/basename
  /opt/ast/bin/cat
  /opt/ast/bin/chmod
  /opt/ast/bin/cmp
  /opt/ast/bin/cut
  /opt/ast/bin/dirname
  /opt/ast/bin/mkdir
  sleep
These tests were backported to improve the test coverage of libcmd.

JohnoKing committed:
Update libcmd tests for pull request

JohnoKing commented:
The pull request has been extended by adding a bunch of backported ksh2020 regression tests for the /opt/ast/bin builtins.

JohnoKing committed:
Increment dates

McDutchie commented on src/lib/libcmd/cp.c:
```diff
@@ -90,7 +90,7 @@ static const char usage_tail[] =
     "responses skip the file.]"
 "[r|R:recursive?Operate on the contents of directories recursively.]"
-"[s:symlink|symbolic-link?Make symbolic links to destination files.]"
+"[s:symlink|symbolic|symbolic-link?Make symbolic links to destination files.]"
```
`--symbolic` already works because getopts completes abbreviated long options, so this addition is redundant.

McDutchie commented on src/cmd/ksh93/tests/builtins.sh:
```diff
@@ -1146,6 +1146,84 @@ fi
+# integer
+# The number of seconds to sleep.
+SECONDS=0
+sleep 1
```
Is this not covered by the test that follows? I am reluctant to add long-sleeping regression tests unless really necessary. Multiple one-second sleeps add up, so I shortened a bunch of them early on. Still, it already takes a long time to run the tests, and the longer they take, the less likely I am to have the time or patience to run them. So, fast regression tests help improve code quality.

JohnoKing replied:
That test is redundant. I have removed it and have also shortened the sleep duration of the other tests.

JohnoKing committed:
Speed up sleep tests, remove redundant cp addition

McDutchie committed:
Fix turning off SHOPT_ALL_LIBCMD

'#ifdef SHOPT_ALL_LIBCMD' is true if it's defined to 0. Since it's
defined as 0 or 1 from SHOPT.sh, it was always treated as on.

McDutchie committed:
regression test tweaks

McDutchie commented:
LGTM now, thanks for the PR!
